src/runtime_intrinsics.c

```diff
@@ -904,7 +904,7 @@ JL_DLLEXPORT jl_value_t *jl_check_top_bit(jl_value_t *a)
     if (!jl_is_bitstype(ty))
         jl_error("check_top_bit: value is not a bitstype");
     if (signbitbyte(jl_data_ptr(a), jl_datatype_size(ty)))
-        jl_throw(jl_inexact_exception);
+        jl_throw(jl_new_struct(jl_inexacterror_type, jl_any_type, jl_any_type, a));
```

Comment:
I assume this depend on how do you want to interpret this? It should be consistent with the one in codegen at least.